### PR TITLE
fix events api snippet, remove dupe "events" variable

### PR DESCRIPTION
**Pull request by josefaidt**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 1]
```diff
@@ -40,56 +40,62 @@ Before you begin, you will need:
...
-  },
+      apiKey: 'da2-abcdefghijklmnopqrstuvwxyz'
+    }
+  }
 });
 
-export default function MyComponent() {
-  const [events, setEvents] = useState<any[]>([]);
+export default function App() {
+  const [myEvents, mySetEvents] = useState<unknown[]>([]);
```
Comment (ErikCH) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:62]: `setMyEvents` <- 👀

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 2]
```diff
@@ -40,56 +40,62 @@ Before you begin, you will need:
...
 
   useEffect(() => {
-    let channel;
+    let channel: EventsChannel;
+    console.log('use effect running');
```
Comment (ErikCH) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:66]: remove

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 3]
```diff
@@ -40,56 +40,62 @@ Before you begin, you will need:
...
+    };
 
     connectAndSubscribe();
 
     return () => channel && channel.close();
   }, []);
 
   async function publishEvent() {
-    await events.post("default/channel", {some: 'data'});
+    console.log('publishing ');
```
Comment (ErikCH) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:86]: remove

**Review by ErikCH — COMMENTED** [review]
Just a few comments (stuff I should have caught)

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx (hunk 1 above)
Comment (josefaidt) [reply src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:62]: oh lol

Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:62]:
```suggestion
  const [myEvents, setMyEvents] = useState<unknown[]>([]);
```

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx (hunk 2 above)
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:66]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx [hunk 4]
```diff
@@ -40,56 +40,61 @@ Before you begin, you will need:
...
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-event-api/index.mdx:85]:
```suggestion
```

**Review by ErikCH — APPROVED** [review]
LGTM

**Review by esauerbo — APPROVED** [review]

**Pull request merged**